include/deal.II/grid/grid_tools.h

```diff
@@ -1727,6 +1727,9 @@ namespace GridTools
                      + cell->face(face)->vertex(2) + cell->face(face)->vertex(3)) / 4.0;
               }
       }
+
+    // Make sure FEValues notices that the mesh has changed
+    triangulation.signals.post_refinement();
```

Comment:
Or maybe a completely separate signal, e.g., `mesh_movement`? Whatever code currently attaches to the `post_refinement` signal may want to also subscribe to the new signal...